### PR TITLE
Adhoc Filters: Fix table filter from mixed datasource panels

### LINKED ISSUE #39386: AdhocFilters: Support mixed data source
**What happened**:

When adding an adhoc filter within a mixed datasource panel, the filter is not getting applied to the queries.

**What you expected to happen**:
Filters are applied to all adhoc enabled datasources within the panel

**How to reproduce it (as minimally and precisely as possible)**:
Create a table panel with a mixed datasource.
Try to filter for a value within the table.

**Anything else we need to know?**:
Looks like the adhoc filter is tied to the mixed datasource type, instead of the concrete datasources used in the panel config

**Environment**:
- Grafana version: v8.1.3 (a61f38238c)

### PR DESCRIPTION
**Pull request by Grubana**
**What this PR does / why we need it**:
Fixes the adhoc filter behaviour on mixed datasource table panels. 
For this to work properly with multiple datasources, the adhoc filters need to be suffixed with its datasource name. 
The label of the adhoc variables can still be changed using the variable editor.

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/grafana/issues/39386

**Special notes for your reviewer**:

### CODE REVIEW

**Review by torkelo — COMMENTED** [review]
Thanks for contributing! 

Not sure supporting this is a great idea (at least given how ad-hoc filters work currently), if you have two different Elasticsearch data sources you will get two ad hoc filters for the same key/value.

**Comment by Grubana** [comment]
Yes thats true.
As far as I understand for now, the adhoc filters of a datasource are replaced if another datasource is filtered.
Theirfore there is no ability to use multiple datasources with adhoc filters in one dashboard.

But I think thats another topic. The main purpose of this PR is to fix the adhoc filter usage with the Mixed datasource.
So i understand that the change I made in the variable creation is pretty out of topic.

I just got this issue with adhoc replacments, when I was using multiple ES datasources in one panel. 

For fixing the main topic of this PR we could also say that just the first adhoc-able datasource gets the filter variable on mixed datasource panels.

Would that be an acceptable solution?

Despite that:
Basically I think it would be better to move the mixed datasource logic into the ```applyFilterFromTable``` method, but that would break the API for custom panels. 

Furthermore, i really like the adhoc functionality and would like to see it in more inbuilt and custom panels and datasources.  
Is there any bigger picture for extending the adhoc functionality?

**Comment by axelavargas** [comment]
Hello @Grubana 👋🏾 , thank you for contributing to the project, unfortunately, it seems we are hesitant on accepting the changes on the PR.  

After checking the original linked issue, and noticing it was moved to a GitHub discussion I am closing this PR. We are going to use that discussion thread to continue building use cases, and additional evidence-based on engagement/voting, in order to help us prioritize this feature request later on down the road.

Thank you 🙏🏾
